ldms/src/core/ldms_stream.c

```diff
@@ -254,6 +256,7 @@ static int __rep_publish(struct ldms_rail_ep_s *rep, const char *stream_name,
 	msg.cred.uid = htobe32(cred->uid);
 	msg.cred.gid = htobe32(cred->gid);
 	msg.perm = htobe32(perm);
+	msg.name_hash = htobe32(hash);
```

Comment:
Nit: Hashes's don't need to be byte-swapped. The are not interpreted by the CPU like integers. This applies to the logic below as well, e.g. even when using the port_no as part of the hash-key, it's just a string of bits at that point.